src/CalcHistoryTraverser.cpp

```diff
@@ -0,0 +1,129 @@
+#include "CalcHistoryTraverser.h"
+
+CalcHistoryTraverser::CalcHistoryTraverser(CalcHistory* p_history)
+{
+    history = p_history;
+    reset();
+};
+
+std::string CalcHistoryTraverser::next()
+{
+    if(curr >= history->newest() || history->isEmpty())
+        return currentInput;
+    if(++curr == history->newest())
+        return currentInput;
+    return curr->getEquation();
+};
+
+std::string CalcHistoryTraverser::current()
+{
+    if(curr >= history->newest() || history->isEmpty())
+        return currentInput;
+    
+    return curr->getEquation();
+};
+
+std::string CalcHistoryTraverser::previous()
+{
+    if(history->isEmpty())
+        return currentInput;
+    if(curr == history->oldest())
+        return curr->getEquation();
+    
+    return (--curr)->getEquation();
+};
+
+void CalcHistoryTraverser::reset()
+{
+    curr = history->newest();
+    currentInput = "";
+};
+
+void CalcHistoryTraverser::setCurrentInput(std::string newInput)
+{
+    curr = history->newest();
+    currentInput = newInput;
+};
+
+std::vector<std::string> CalcHistoryTraverser::getHistoryStringWithBounds(int prevSize, int nextSize, int width)
+{
+    std::vector<std::string> reversePrev;
+    if(!history->isEmpty())
+        for(std::vector<CalcHistoryPair>::iterator it = curr - 1; it >= history->oldest() && (curr - it) < prevSize; it--)
+            reversePrev.push_back(it->getEquation());
```

Comment:
Is there a reason that this push_back doesn't enforce the same width requirement as the forwardNext.push_back below?